update_cpes.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+
+
+import json
+import logging
+import re
+import sys
+
+from lxml import etree
+
+def parse_r7_remapping(file):
+    remap = {} # r7_vendor => { 'cpe_vendor' => <cpe_vendor>, 'products': { r7_product1 => cpe_product1 }}
+    remappings = None
+    with open(file) as remap_file:
+        remappings = json.load(remap_file)["remappings"]
+
+    for remap_json in remappings:
+        r7_vendor = remap_json['r7_vendor']
+        cpe_vendor = remap_json['cpe_vendor']
+        if r7_vendor in remap:
+            raise ValueError("R7 vendor {} duplicated in {}".format(r7_vendor, file))
+
+        product_map = {}
+        if 'products' in remap_json:
+            product_map = remap_json['products']
+        remap[r7_vendor] = {'cpe_vendor': cpe_vendor, 'products': product_map}
+
+    return remap
+
+
+def parse_cpe_vp_map(file):
+    vp_map = {} # cpe_type -> vendor -> products
+    parser = etree.XMLParser(remove_comments=False)
+    doc = etree.parse(file, parser)
+    namespaces = {'ns': 'http://cpe.mitre.org/dictionary/2.0', 'meta': 'http://scap.nist.gov/schema/cpe-dictionary-metadata/0.2'}
+    for meta in doc.xpath("//ns:cpe-list/ns:cpe-item/meta:item-metadata[@status = 'FINAL']", namespaces=namespaces):
+        final_cpe_name = meta.getparent().attrib['name']
+
+        cpe_match = re.match('^cpe:/([aho]):([^:]+):([^:]+)', final_cpe_name)
+        if cpe_match:
+            cpe_type, vendor, product = cpe_match.group(1, 2, 3)
+            if not cpe_type in vp_map:
+                vp_map[cpe_type] = {}
+            if not vendor in vp_map[cpe_type]:
+                vp_map[cpe_type][vendor] = set()
+            vp_map[cpe_type][vendor].add(product)
+        else:
+            logging.error("Unexpected CPE %s", final_cpe_name)
+
+    return vp_map
```

Comment:
May want to add a sanity check here.  If no records are imported from the XML then log an error and exit.  Ran into this because it seems that I don't have the correct XML file.